zenodo/modules/records/views.py

```diff
@@ -354,6 +354,31 @@ def record_from_pid(recid):
         return {}
 
 
+@blueprint.app_template_filter('sort_custom_keywords')
+def sort_custom_keywords(keywords):
+    custom_keywords_config = current_app.config.get(
+            'ZENODO_CUSTOM_METADATA_VOCABULARIES')
+
+    sortable_keywords = []
+    unsorted_keywords = []
+
+    for keyword in keywords:
+        (key, value) = keyword
+        (category, index) = key.split(':')
+        if category in custom_keywords_config.keys():
+            attribute = custom_keywords_config[category]['attributes'][index]
+            if 'order' in attribute.keys():
+                modified_keyword = (keyword[0], keyword[1], attribute['order'])
+                sortable_keywords.append(modified_keyword)
+            else:
+                unsorted_keywords.append(keyword)
+
+    sortable_keywords.sort(key=lambda x: x[2])
+    sorted_keywords = list(map(lambda x: (x[0], x[1]), sortable_keywords))
```

Comment:
```suggestion
    sorted_keywords = [(term, values) for term, values, _ in sortable_keywords]
```
Not to be applied for this block necessarily, but just to show a Pythonic way for transforming lists (instead of using `map`). This is a list comprehension and allows for transforming and filtering lists. I would recommend [this introduction](https://treyhunner.com/2015/12/python-list-comprehensions-now-in-color/) and [the follow-up post](https://treyhunner.com/2019/03/abusing-and-overusing-list-comprehensions-in-python/), to get a hang of it.